source/common/common/utility.h

```diff
@@ -779,6 +778,24 @@ template <class Value> class TrieLookupTable {
     return result ? result->value_ : nullptr;
   }
 
+  ~TrieLookupTable() {
+    // To avoid stack overflow on recursive destruction if the tree is very deep,
+    // flatten the tree first.
```

Comment:
Is the following flattening the entire tree or just 2 levels at a time?

Reply:
I think it is flattening the entire tree like BFS coupling the vector push back.

Reply:
Ah, I missed that the second for loop adds the elements to the list, and the iteration is over the number of elements.
Thanks!